tools/cli/tests/end-to-end.py

```diff
@@ -171,8 +185,8 @@ def test_connect(self):
             self.assertIn('TERMINATED', instance.status())
             with tempfile.NamedTemporaryFile() as tmp:
                 with DatalabConnection(self.project, self.zone,
-                                       instance.name, tmp):
-                    readme = urlopen(readme_url)
+                                       instance.name, tmp) as conn:
+                    readme = urlopen(conn.readme_url)
```

Comment:
It seems there are some cases where this wouldn't be defined?

Reply:
Yes; if the connection fails. In that case we want to raise an exception, so I figured it was fine to just let it raise the exception here.

Reply:
I see, lgtm then.